File: src/iso.image.ts

import { BaseEntity, ScreenPosition, TileMap } from './iso'

export class Image extends BaseEntity {

    constructor(  private basename:string, screenPos:ScreenPosition, private map:TileMap) {
        super(screenPos)
    }

    render():void {
        this.map.renderImage( this.basename, this.screenPos )
    }


}

File: rust-isomap/src/iso.image.rs
mod iso;

struct Image {
    screen_pos: Pos,
    basename: &str,
    map: &TileMap,
}

impl Entity for Image {

    fn screen_pos(&self) -> &Pos {
        &self.screen_pos
    }
    
    fn render(&self) {
        self.map.renderImage( self.basename, self.screen_pos )
    }

}

